cmd/main.go

```diff
@@ -154,31 +161,22 @@ func main() {
 		// wait 2 seconds to allow reconciling webhookconfiguration and service endpoint.
 		time.Sleep(2 * time.Second)
 
-		if err = featuregates.ParseAndValidateFeatureGates(featureGates); err != nil {
+		if err := featuregates.ParseAndValidateFeatureGates(featureGates); err != nil {
 			klog.ErrorS(err, "unable to set `feature-gates` flag")
 			exitWithErrorFunc()
 		}
 	}
 
+	err = nodeclaim.CheckNodeClass(ctx, kClient)
```

Comment:
Why do you remove the feature gate check here?

Reply:
Because it is redundant in this case. The nodeclaim won't get triggered unless the feature gate flag is on

Reply:
Fixed.